controls/radvirtualgrid/mvvm.md

```diff
@@ -0,0 +1,327 @@
+                }
+            }
+        }
+
+        public DateTime Established
+        {
+            get { return this.established; }
+            set
+            {
+                if (value != this.established)
+                {
+                    this.established = value;
+                    this.OnPropertyChanged("Established");
+                }
+            }
+        }
+
+        public int StadiumCapacity
+        {
+            get { return this.stadiumCapacity; }
+            set
+            {
+                if (value != this.stadiumCapacity)
+                {
+                    this.stadiumCapacity = value;
+                    this.OnPropertyChanged("StadiumCapacity");
+                }
+            }
+        }
+
+        public Club()
+        {
+
+        }
+
+        public Club(string name, DateTime established, int stadiumCapacity)
+        {
+            this.name = name;
+            this.established = established;
+            this.stadiumCapacity = stadiumCapacity;
+        }
+
+        protected virtual void OnPropertyChanged(PropertyChangedEventArgs args)
+        {
+            PropertyChangedEventHandler handler = this.PropertyChanged;
+            if (handler != null)
+            {
+                handler(this, args);
+            }
+        }
+
+        private void OnPropertyChanged(string propertyName)
+        {
+            this.OnPropertyChanged(new PropertyChangedEventArgs(propertyName));
+        }
+
+        public override string ToString()
+        {
+            return this.Name;
+        }
+
+        public static ObservableCollection<Club> GetClubs()
+        {
+            ObservableCollection<Club> clubs = new ObservableCollection<Club>();
+            Club club;
+            int id = 0;
+
+            // Liverpool
+            club = new Club("Liverpool", new DateTime(1892, 1, 1), 45362) { PrimaryKey = id};
+            clubs.Add(club);
+
+            id++;
+            // Manchester Utd.
+            club = new Club("Manchester Utd.", new DateTime(1878, 1, 1), 76212) {PrimaryKey = id};
+            clubs.Add(club);
+
+            id++;
+            // Chelsea
+            club = new Club("Chelsea", new DateTime(1905, 1, 1), 42055) { PrimaryKey = id };
+            clubs.Add(club);
+
+            id++;
+            // Arsenal
+            club = new Club("Arsenal", new DateTime(1886, 1, 1), 60355) { PrimaryKey = id };
+            clubs.Add(club);
+
+            return clubs;
+        }
+    }
+{{endregion}}
+
+## Implementing a custom DataProvider
+
+Then, a custom __DataProvider__ that handles some specific scenarios will be defined. Speaking shortly, it demonstrates how the selection of the control can be persisted when sorting or filtering. Note, that by default it is being cleared when processing these operations. Also, hiding a given column, defining a custom editor, setting a given column to be readonly and applying a custom header is demonstrated.
```

Comment:
Here you can start the new section with something more describing, or starting, or welcoming, I am not sure what word to use. Currently, the word 'Then' is hanging into the void.  This is a new section and it should start so that you can read only this and get the information that section states it provides in its title.

For example, in this case you can start with something like "The next step is to implement a custom DataProvider that handles some specific scenarios." or "This section describes how to implement a custom DataProvider.."

Remove "speaking shortly" from the second sentence. It is quite formal phrase.

"Note, that tby default it is being cleared..."  - what has been cleared? What is this 'it'? Use the word here because it is not very clear what does it mean.

"Also, hiding a given column, defining a custom editor, setting a given column to be readonly and applying a custom header is demonstrated."  - the word "demonstrated" is way too far and its meaning is a bit blurry. Can we move it in the beginning of the sentence. For example "Also, the example demonstrates how to hide a given column, define a column editor, etc."